### PR TITLE
refactor: convert ipc-renderer.ts to TypeScript

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Convert `ipc-renderer.ts` to TypeScript. Also make the `ipc` binding typesafe.

Note: there is an issue with `ipcRenderer.sendToAll`, which is not documented. We can either remove it or document it.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: Removed undocumented `ipcRenderer.sendToAll()`.

### CODE REVIEW

File: lib/renderer/api/ipc-renderer.ts [hunk 1]
```diff
@@ -23,15 +21,9 @@ ipcRenderer.sendTo = function (webContentsId, channel, ...args) {
   return ipc.sendTo(internal, false, webContentsId, channel, args)
 }
 
-ipcRenderer.sendToAll = function (webContentsId, channel, ...args) {
```
Comment (nornagon) [inline lib/renderer/api/ipc-renderer.ts]: I'm pretty sure we use this for our chrome extension stuff.

Comment (miniak) [reply lib/renderer/api/ipc-renderer.ts]: we use `ipcRendererInternal.sendToAll()` there

Comment (miniak) [reply lib/renderer/api/ipc-renderer.ts]: https://github.com/electron/electron/blob/master/typings/internal-electron.d.ts#L65-L67

Comment (miniak) [reply lib/renderer/api/ipc-renderer.ts]: this is about the public API

Comment (miniak) [reply lib/renderer/api/ipc-renderer.ts]: @MarshallOfSound do you think the public API should be documented instead being removed as undocumented?

File: typings/internal-ambient.d.ts [hunk 2]
```diff
@@ -14,6 +14,14 @@ declare namespace NodeJS {
     isComponentBuild(): boolean;
   }
 
+  interface IpcBinding {
+    send(internal: boolean, channel: string, args: any[]): void;
+    sendSync(internal: boolean, channel: string, args: any[]): any;
+    sendToHost(channel: string, args: any[]): void;
+    sendTo(internal: boolean, sendToAll: boolean, webContentsId: number, channel: string, args: any[]): void;
+    invoke(channel: string, args: any[]): Promise<{ error: string, result: any }>;
```
Comment (MarshallOfSound) [inline typings/internal-ambient.d.ts]:
```suggestion
    invoke<T>(channel: string, args: any[]): Promise<{ error: string, result: T }>;
```

**Comment by ConradIrwin** [comment]
@miniak We are currently using ipcRenderer.sendToAll() in our electron app at Superhuman to emulate broadcast channels.

Is there a replacement and/or can we reimplement it ourselves easily?

**Pull request merged**